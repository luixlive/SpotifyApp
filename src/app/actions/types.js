// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
