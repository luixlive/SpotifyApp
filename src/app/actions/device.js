import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
