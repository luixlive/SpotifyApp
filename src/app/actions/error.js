import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
