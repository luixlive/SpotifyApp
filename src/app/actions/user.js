import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
