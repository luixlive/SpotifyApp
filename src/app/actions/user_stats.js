import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
