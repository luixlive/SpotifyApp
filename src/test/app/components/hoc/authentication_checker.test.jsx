import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { MemoryRouter, Route, Switch } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';

import addRouterProps from './../../../test_utils/add_router_props';
import {
  authenticationChecker as componentProps,
} from './../../../test_utils/components_props';
import AuthenticationChecker, {
  getAuthenticationChecker,
} from './../../../../app/components/hoc/authentication_checker';
import initialState from './../../../test_utils/initial_state';
import * as types from './../../../../app/actions/types';

describe('App Components HOC - AuthenticationChecker', () => {
  describe('Snapshots', () => {
    let AuthenticationCheckerComponent;
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
      AuthenticationCheckerComponent = getAuthenticationChecker(() => <div />);
    });

    it('renders user not authenticated', () => {
      const rendered =
        renderer.create(<AuthenticationCheckerComponent {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders Redirect if user not authenticated and load finished', () => {
      props.loadUserFinished = true;
      const rendered = mount((
        <MemoryRouter initialIndex={0} initialEntries={['/stats']} >
          <Switch>
            <Route exact path="/" component={() => <div>Should render this</div>} />
            <Route path="/stats" component={() => <AuthenticationCheckerComponent {...props} />} />
          </Switch>
        </MemoryRouter>
      ));
      expect(toJson(rendered.find('div'))).toMatchSnapshot();
    });

    it('renders user authenticated and load finished', () => {
      props.isUserAuthenticated = true;
      props.loadUserFinished = true;
      const rendered =
        renderer.create(<AuthenticationCheckerComponent {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders user authenticated and logging out', () => {
      props.isUserAuthenticated = true;
      props.loadUserFinished = true;
      props.loggingOutUser = true;
      const rendered =
        renderer.create(<AuthenticationCheckerComponent {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    let AuthenticationCheckerComponent;
    let props;
    beforeEach(() => {
      props = _.cloneDeep(componentProps);
      props.loadUser = jest.fn();
      AuthenticationCheckerComponent = getAuthenticationChecker(() => <div />);
    });

    it('should call loadUser if user not authenticated', () => {
      mount(<AuthenticationCheckerComponent {...props} />);

      expect(props.loadUser).toHaveBeenCalledTimes(1);
    });

    it('shouldnt call loadUser if user authenticated', () => {
      props.isUserAuthenticated = true;
      mount(<AuthenticationCheckerComponent {...props} />);

      expect(props.loadUser).toHaveBeenCalledTimes(0);
    });
  });

  describe('Provider', () => {
    let RenderedComponent;
    let store;
    let wrapper;
    beforeAll(() => {
      const ComposedComponent = () => <div>Component</div>;
      const mockStore = configureStore();
      store = mockStore(initialState);
      RenderedComponent = AuthenticationChecker(ComposedComponent);
      wrapper = mount((
        <MemoryRouter>
          <Provider store={store}>
            <RenderedComponent />
          </Provider>
        </MemoryRouter>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(RenderedComponent).length).toEqual(1);
    });

    it('matches initial state', () => {
      const props = addRouterProps(componentProps);
      expect(Object.keys(wrapper.find('AuthenticationChecker').props()))
        .toEqual(Object.keys(props));
    });

    it('dispatches loadUser', () => {
      const expectedAction = {
        type: types.LOAD_USER,
        payload: { },
      };
      expect(store.getActions()[0]).toEqual(expectedAction);
    });
  });
});
