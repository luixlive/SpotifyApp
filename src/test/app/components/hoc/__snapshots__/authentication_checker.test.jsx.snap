// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components HOC - AuthenticationChecker Snapshots renders user authenticated 1`] = `<div />`;

exports[`App Components HOC - AuthenticationChecker Snapshots renders user authenticated and logging out 1`] = `
<div
  className="ui basic segment body"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted text loader"
        >
          <h3
            className="ui header"
          >
            Loading...
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Components HOC - AuthenticationChecker Snapshots renders user not authenticated 1`] = `
<div>
  Should render this
</div>
`;
